### ISSUE TITLE
Project name with dot [.] at the end causes folder lockup on Windows

### ISSUE THREAD

**Issue by Banderi** [godotengine/godot#32330]
**Godot version:**
Both 3.1 and 3.2.alpha that I tried so far

**OS/device including version:**
Windows 7 Professional 64-bit

**Issue description:**
Saving a project that ends with a dot (.) on Windows will result in some locations, e.g. the `user://` directory, to be treated as corrupted paths and become unavailable and unremovable.

**Steps to reproduce:**
1) Start a new project
2) Save project with a name that ends with a dot
3) Notice how, if you try to write/read to `user://` GDScript will quietly fail with such errors:
```
E 0:00:26:0731   Condition ' !f ' is true.
  <C Source>     core/bind/core_bind.cpp:2128 @ store_var()
  <Stack Trace>  game.gd:274 @ savetofile()
                 game.gd:322 @ save_slotdata()
                 game.gd:330 @ save_game()
                 game.gd:353 @ _ready()
```
4: Navigate to %AppData%/Godot/app_userdata/ and notice your folder [cannot be removed by conventional means](https://stackoverflow.com/questions/4075753/how-to-delete-a-folder-that-name-ended-with-a-dot)

Naturally I don't think this is strictly Godot's fault, rather Windows for being dumb with files and folder names; however I felt like it was worth mentioning, and probably would be a good idea to disallow such project names (or tweak how they are handled internally)

**Comment by Calinou** [comment]
Out of curiosity, does this occur with other characters such as `?` or `!`?

**Comment by Banderi** [comment]
So far that I can see, only dots cause issues.
Appending more than one dot (with no other characters after them), slashes/backslashes or fully empty project names are ignored and won't even create any folder. Question marks, apparently are converted in an ending hyphen (`-`) at least on my machine/OS version/locale, and any other character so far seems to work normally.

**Comment by Prime-Eight** [comment]
windows is an extension sensitive OS. ther are rules particular to windows as for what characters you can append a file and folder with, especially after a '.' .
dots are part of the shell/dos system on windows. along with a bunch of the punctuation characters, 
when used in a file/folder name some of them indented uses, and some just mess up the shell.
Some malware exploits this by naming file sin way the make them require you to first find them ( they won't show in the explorer any more) and then use the shell directly to rename or remove them.
It's not all bad though, some OS and system files and areas are kept hidden properly using simple naming of folders and files. 

" Danger zone! If you direly want a folder name to end with ., you'll need to use the magic raw name override sequence of \\?\. In a command prompt, md \\?\C:\path\to\container\._. will indeed create a folder named ._., but lots of programs will have problems with it, even Explorer:
._. problems

Such a directory can only be removed with rd followed by its \\?\ name or renamed with its short (8.3, dir /x) name.
" by: Ben N https://superuser.com/users/380318/ben-n

for dos/windows, the period was only meant to separate a base file name from an extension, and that legacy continues.

" from link below 
Do not end a file or directory name with a space or a period. Although the underlying file system may support such names, the Windows shell and user interface does not. However, it is acceptable to specify a period as the first character of a name. For example, ".temp".
"
There is a bunch of reserved names as well . 
https://docs.microsoft.com/en-ca/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN

**Comment by KoBeWi** [comment]
Still valid in db5ea78b7

**Comment by Sslaxx** [comment]
It's hell to try and delete a file or directory that ends in a full stop on Windows. So this kind of behaviour isn't unexpected, alas, and might only be resolvable by explicitly forbidding the use of `.` at the end of a directory path or file name.

**Comment by Calinou** [comment]
Project names ending with a period are already supposed to be forbidden: https://github.com/godotengine/godot/blob/0cee3021927bc246a42c589cc28199a4c51f2909/editor/project_manager.cpp#L380

Is something not working as expected here? Should it be an error instead of a warning?

**Comment by KoBeWi** [comment]
Seems fixed in 69e5ad7
The dot is automatically removed when creating/opening directory.
